sunpy/map/header_helper.py

```diff
@@ -37,7 +37,8 @@ def make_fitswcs_header(data, coordinate,
                         observatory=None,
                         wavelength: u.angstrom = None,
                         exposure: u.s = None,
-                        projection_code="TAN"):
+                        projection_code="TAN",
+                        unit=None):
```

Comment:
Just to complicate this, what if `data` is a quantity? Should we use that unit?

Reply:
I like that idea...but I wonder if it would be confusing given the fact that you cannot initiate a map with a `Quantity` (or at least, we don't explicitly support that).

Reply:
Ok, I've added the option to do this as well. It's pretty trivial. I've taken the NDCube approach of the `unit=` kwarg taking precedence if units are specified on `data` as well. This gets logged to `INFO` if that's the case.

Reply:
@dstansby you may want to have another peek at this since you approved before this addition.